components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/CarbonAppResource.java

```diff
@@ -56,18 +78,128 @@ public boolean invoke(MessageContext messageContext) {
+            JSONObject response = Utils.createJsonError("Supports only for the Content-Type : "
+                    + MULTIPART_FORMDATA_DATA_TYPE, axisMsgCtx, BAD_REQUEST);
+            Utils.setJsonPayLoad(axisMsgCtx, response);
+            return;
+        }
+
+        StringBuilder unMovedCApps = new StringBuilder();
+        String errorMessage = "Error when deploying the Carbon Application : ";
+        OMElement messageBody = axisMsgCtx.getEnvelope().getBody().getFirstElement();
+        Iterator iterator = messageBody.getChildElements();
+        boolean isDeployedSuccesfully = true;
+        while (iterator.hasNext()) {
+            OMElement fileElement = (OMElement) iterator.next();
+            String fileName = fileElement.getAttributeValue(new QName("filename"));
+            if (fileName != null && fileName.endsWith(".car")) {
+                byte[] bytes = Base64.getDecoder().decode(fileElement.getText());
+                Path cAppDirectoryPath = Paths.get(Utils.getCarbonHome(), "repository", "deployment",
+                        "server", "carbonapps", fileName);
+                try {
+                    Files.write(cAppDirectoryPath, bytes);
+                } catch (IOException e) {
+                    isDeployedSuccesfully = false;
+                    unMovedCApps.append(fileName).append(", ");
+                    log.error(errorMessage + fileName, e);
+                }
+                log.info("Successfully added Carbon Application : " + fileName);
+            } else {
+                isDeployedSuccesfully = false;
+                unMovedCApps.append((fileName != null ? fileName : "filename: <null>")).append(", ");
+            }
+        }
+
+        if (isDeployedSuccesfully) {
+            jsonResponse.put(Constants.MESSAGE_JSON_ATTRIBUTE, "Successfully added Carbon Application(s)");
+            Utils.setJsonPayLoad(axisMsgCtx, jsonResponse);
+        } else {
+            Utils.setJsonPayLoad(axisMsgCtx, Utils.createJsonErrorObject(errorMessage + unMovedCApps));
+        }
+    }
+
+    private void handleDelete(MessageContext messageContext, org.apache.axis2.context.MessageContext axisMsgCtx) {
+        String cAppNamePattern = Utils.getPathParameter(messageContext, CAPP_NAME_PATTERN);
+        JSONObject jsonResponse = new JSONObject();
+        if (!Objects.isNull(cAppNamePattern)) {
+            try {
+                String cAppsDirectoryPath = Paths.get(
+                        Utils.getCarbonHome(), "repository", "deployment", "server", "carbonapps").toString();
+
+                // List deployed CApps which has downloaded CApp name prefix
+                File carbonAppsDirectory = new File(cAppsDirectoryPath);
+                File[] existingCApps = carbonAppsDirectory.listFiles(new FilenameFilter() {
+                    public boolean accept(File dir, String name) {
+                        return name.contains(cAppNamePattern) && name.endsWith(".car");
```

Comment:
I think the logic should change. Suppose we have 3 CApps deployed, test, test1 and test2. If you send a delete request with 'test' this will delete all 3 Capps.

Reply:
ya changed to name.equals(cAppName) && name.endsWith(".car")